### PR TITLE
feat: Select all permissions button for subusers

### LINKED ISSUE #3851: Select all permissions with one button
### Is there an existing feature request for this?

- [X] I have searched the existing issues before opening this feature request.

This looks like a duplicate of #2383, but it's not. In that issue this button was mentioned, but the closing PR did not also add the select all button requested, only the Save button.

### Describe the feature you would like to see.

Add a button on top to select all the permissions with one click. There are a lot of categories, and if you want to add multiple subusers it's easier to add all the permissions using one button.

### Describe the solution you'd like.

Add a button to select all permission categories with one click, like in the v0.x builds.

### Additional context to this request.
image

*(Thanks to #2464 for the image)*

### PR DESCRIPTION
**Pull request by theMackabu**
This solves issue #3851 
Idea was based off #3869 
_Components tested with multiple views, permission assignment also tested._

SelectionBoxes

All Selected

Sorry about the auto-format of the code

### CODE REVIEW

**Comment by trenutoo** [comment]
Please use eslint and set it as formatter to match the code style. This PR changes the whole code style of every code line, e.g., indentation, single vs. double quotes

File: resources/scripts/components/server/users/EditSubuserModal.tsx [hunk 1]
```diff
@@ -105,45 +110,49 @@ const EditSubuserModal = ({ subuser }: Props) => {
...
-                                    key={`permission_${key}.${pkey}`}
-                                    permission={`${key}.${pkey}`}
-                                    disabled={!canEditUser || editablePermissions.indexOf(`${key}.${pkey}`) < 0}
-                                />
-                            ))}
-                        </PermissionTitleBox>
-                    ))}
+                    <div css={tw`flex items-center mb-4 bg-neutral-600 p-2 rounded shadow-sm`}>
+                        <p css={tw`text-sm flex-1 ml-1 text-neutral-200`}>Select all permissions</p>
+                        {canEditUser && <PermissionSelectAll isEditable={canEditUser} permissions={editablePermissions}></PermissionSelectAll>}
```
Comment (OpenSource03) [inline resources/scripts/components/server/users/EditSubuserModal.tsx]: Collapse this open empty tag. <PermissionSelectAll></PermissionSelectAll>

**Comment by DaneEveritt** [comment]
@theMackabu I am good to merge this, but please undo all of the changes to formatting that were made. There is a style enforced on the codebase and this PR introduces a lot of completely unrelated spacing and formatting changes.

**Comment by WavingCatDevs** [comment]
I would love this to be merged, always wanted this feature.

**Comment by theMackabu** [comment]
closing pr, this is pretty stale and could be done better by me

**Pull request closed without merging**